### PR TITLE
Trivial none return type

### PR DESCRIPTION
**Pull request by hjmjohnson**
Related to #476, #484 

Depends on #499 

### Description
Add `-> None` to functions to explicitly indicate they do not have return values.

Add enumeration for consistency of interp_order.

### Status
Ready

### Types of changes
- [x] Docstrings/Documentation updated

### CODE REVIEW

**Comment by hjmjohnson** [comment]
This previously passed all tests.  It would be nice to keep these type-hinting items flowing in so that we can focus on completing this task.

File: monai/handlers/checkpoint_saver.py [hunk 1]
```diff
@@ -140,34 +140,44 @@ def attach(self, engine: Engine):
...
-    def completed(self, engine):
+    def completed(self, engine) -> None:
         """Callback for train or validation/evaluation completed Event.
         Save final checkpoint if configure save_final is True.
 
         """
+        assert callable(self._final_checkpoint), "Error: _final_checkpoint function not specified."
         self._final_checkpoint(engine, self.save_dict)
+        assert self.logger is not None
+        assert hasattr(self.logger, "info"), "Error, provided logger has not info attribute."
```
Comment (wyli) [inline monai/handlers/checkpoint_saver.py:151]: trying to fix this logger in #539, could you have a look?

**Comment by hjmjohnson** [comment]
@hjmjohnson I verified that the asserts are still needed.  Could this outstanding PR be merged?

**Comment by Nic-Ma** [comment]
> @hjmjohnson I verified that the asserts are still needed. Could this outstanding PR be merged?

seems CI failed for some error.

**Comment by benjamin-gorman** [comment]
@Nic-Ma @wyli

**Pull request merged**